integrations/pinecone/src/pinecone_haystack/document_store.py

```diff
@@ -0,0 +1,236 @@
+        """
+        try:
+            count = self._index.describe_index_stats()["namespaces"][self.namespace]["vector_count"]
+        except KeyError:
+            count = 0
+        return count
+
+    def write_documents(self, documents: List[Document], policy: DuplicatePolicy = DuplicatePolicy.NONE) -> int:
+        """
+        Writes Documents to Pinecone.
+
+        :param documents: A list of Documents to write to the document store.
+        :param policy: The duplicate policy to use when writing documents.
+            PineconeDocumentStore only supports `DuplicatePolicy.OVERWRITE`.
+
+        :return: The number of documents written to the document store.
+        """
+        if len(documents) > 0:
+            if not isinstance(documents[0], Document):
+                msg = "param 'documents' must contain a list of objects of type Document"
+                raise ValueError(msg)
+
+        if policy not in [DuplicatePolicy.NONE, DuplicatePolicy.OVERWRITE]:
+            logger.warning(
+                f"PineconeDocumentStore only supports `DuplicatePolicy.OVERWRITE`"
+                f"but got {policy}. Overwriting duplicates is enabled by default."
+            )
+
+        documents_for_pinecone = []
+        for document in documents:
+            if document.embedding is None:
+                logger.warning(
+                    f"Document {document.id} has no embedding. Pinecone is a purely vector database. "
+                    "A dummy embedding will be used, but this can affect the search results. "
+                )
+                document.embedding = self._dummy_vector
+            doc_for_pinecone = {"id": document.id, "values": document.embedding, "metadata": document.meta}
+
+            # we save content/dataframe as metadata
+            # currently, storing blob in Pinecone is not supported
+            if document.content is not None:
+                doc_for_pinecone["metadata"]["content"] = document.content
+            if document.dataframe is not None:
+                doc_for_pinecone["metadata"]["dataframe"] = document.dataframe.to_json()
+            if document.blob is not None:
+                logger.warning(
+                    f"Document {document.id} has a blob. Currently, storing blob in Pinecone is not supported. "
+                    "The blob will be ignored."
+                )
+
+            documents_for_pinecone.append(doc_for_pinecone)
+
+        result = self._index.upsert(
+            vectors=documents_for_pinecone, namespace=self.namespace, batch_size=self.batch_size
+        )
+
+        written_docs = result["upserted_count"]
+        return written_docs
+
+    def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Document]:
+        if not filters:
+            # in this case, we try to return all documents but Pinecone has some limits
+            documents = self._embedding_retrieval(query_embedding=self._dummy_vector, top_k=TOP_K_LIMIT)
```

Comment:
nit: this made me go and check why you were not passing `namespace` (while you do pass `top_k`). I would probably pass `namespace` to `_embedding_retrieval` to ease readability